mlir-tensorrt/executor/lib/Runtime/Backend/Lua/Modules/TensorRT/TensorRTModule.cpp

```diff
@@ -198,6 +198,195 @@ class NvInferExecContextWrapper {
   /// the TRT network.
   std::vector<PinnedMemoryBlock> hostIOBuffers;
 };
+
+/// OutputAllocatorImpl - Implementation of TensorRT's IOutputAllocator
+/// interface. This class manages memory allocation for TensorRT outputs.
+class OutputAllocatorImpl : public nvinfer1::IOutputAllocator {
+public:
+  OutputAllocatorImpl() = default;
+  /// It is the responsibilty of runtime client to release the buffer.
+  ~OutputAllocatorImpl() {}
+
+  // Disable copy and move operations
+  OutputAllocatorImpl(const OutputAllocatorImpl &) = delete;
+  OutputAllocatorImpl &operator=(const OutputAllocatorImpl &) = delete;
+  OutputAllocatorImpl(OutputAllocatorImpl &&) = delete;
+  OutputAllocatorImpl &operator=(OutputAllocatorImpl &&) = delete;
+
+  void setTensorName(const std::string &tensorName) {
+    mTensorName = tensorName;
+  }
+  void setCurrentMemory(void *currentMemory, int64_t size) {
+    mCurrentMemory = currentMemory;
+    mOutputSize = size;
+  }
+
+  // IOutputAllocator interface implementation
+  void *reallocateOutputAsync(const char *tensorName, void *currentMemory,
+                              uint64_t size, uint64_t alignment,
+                              cudaStream_t stream) override {
+    assert(currentMemory == mCurrentMemory && "Output buffer mismatch");
+    assert(tensorName == mTensorName && "Tensor name mismatch");
+    assert(!mReallocateOutputCalled && "Duplicate call to reallocateOutput");
+    mReallocateOutputCalled = true;
+
+    size = std::max(size, static_cast<uint64_t>(1));
+
+    if (size > mOutputSize) {
+      size = roundUp(size, alignment);
+
+      if (mOutputPtr) {
+        cudaFree(mOutputPtr);
+      }
+
+      mOutputPtr = nullptr;
+      mOutputSize = 0;
+
+      void *newMemory;
+      if (cudaMalloc(&newMemory, size) == cudaSuccess) {
```

Comment:
Any allocation needs to be  made via the `AllocTracker`, otherwise the runtime client won't have visibility